src/v/redpanda/admin_server.cc

```diff
@@ -622,16 +623,14 @@ void admin_server::log_exception(
 void admin_server::rearm_log_level_timer() {
     _log_level_timer.cancel();
 
-    auto next = std::min_element(
-      _log_level_resets.begin(),
-      _log_level_resets.end(),
-      [](const auto& a, const auto& b) {
-          return a.second.expires < b.second.expires;
-      });
-
-    if (next != _log_level_resets.end()) {
-        _log_level_timer.arm(next->second.expires);
+    if (_log_level_resets.empty()) {
+        return;
     }
+
+    auto reset_values = _log_level_resets | std::views::values;
+    auto& lvl_rst = *std::ranges::min_element(
+      reset_values, std::less<>{}, &level_reset::expires);
+    _log_level_timer.arm(lvl_rst.expires);
```

Comment:
Is the original actually bugged? Seems like the motivating issue would have been isolated to v23.3+

Reply:
ah you are right, i didn't notice that v23.2 has the correct comparator